client/src/components/Draftail/index.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { DraftailEditor } from 'draftail';
+
+import { IS_IE11 } from '../../config/wagtailConfig';
+
+import Icon from '../Icon/Icon';
+
+import sources from './sources';
+import Link from './decorators/Link';
+import Document from './decorators/Document';
+import ImageBlock from './blocks/ImageBlock';
+import EmbedBlock from './blocks/EmbedBlock';
+
+import registry from './registry';
+
+const wrapWagtailIcon = type => {
+  if (type.icon) {
+    return Object.assign(type, {
+      icon: <Icon name={type.icon} />,
+    });
+  }
+
+  return type;
+};
+
+export const initEditor = (fieldName, options = {}) => {
+  const field = document.querySelector(`[name="${fieldName}"]`);
+  const editorWrapper = document.createElement('div');
+  field.parentNode.appendChild(editorWrapper);
+
+  const serialiseInputValue = rawContentState => {
+    // TODO Remove default {} when finishing https://github.com/springload/wagtaildraftail/issues/32.
+    field.value = JSON.stringify(rawContentState || {});
```

Comment:
I believe it was in anticipation of springload/wagtaildraftail#32 where we wanted the default to be `None` instead of `{}` on the Python side.

That being said, your second link shows that `value` might still be `None`:
https://github.com/gasman/wagtail/blob/1b4ff77034f03a94f141d44884d921d4f3c99106/wagtail/admin/rich_text/editors/draftail/__init__.py#L47-L51